<launch>
    <arg name="use_ground_truth_state" default="false"/>
    <arg name="use_guidance" default="false"/>
    <arg name="use_fixed_guidance" default="false"/>
    <arg name="use_tracking_controller" default="NONE"/>
    <arg name="tracking_type" default="state"/>

    <group ns="control/mpc" unless="$(arg use_guidance)">
        <rosparam file="$(find drone_optimal_control)/config/mpc_config.yaml"/>
    </group>

    <group ns="control/mpc" if="$(arg use_guidance)">
        <rosparam file="$(find drone_optimal_control)/config/mpc_config_guidance_tracking.yaml"/>
    </group>

    <group ns="control/mpc" if="$(arg use_fixed_guidance)">
        <rosparam file="$(find drone_optimal_control)/config/mpc_config_fixed_guidance_tracking.yaml"/>
    </group>

    <group ns="guidance/mpc">
        <rosparam file="$(find drone_optimal_control)/config/guidance_config.yaml"/>
    </group>

<!--    valgrind &#45;&#45;track-origins=yes &#45;&#45;leak-check=full-->

    <group if="$(arg use_guidance)">
        <node name="guidance" pkg="drone_optimal_control" type="drone_guidance" output="screen"
              launch-prefix="taskset -c 2">
            <rosparam param="target_apogee">[0, 0, 4]</rosparam>
            <rosparam param="target_land">[0, 1, 0]</rosparam>
        </node>
    </group>

    <group if="$(arg use_fixed_guidance)">
        <node name="guidance" pkg="drone_optimal_control" type="fixed_guidance" cwd="node" output="screen">
            <rosparam param="speed">0.35</rosparam>
            <rosparam param="scaling">1</rosparam>
            <rosparam param="rotation_angle">0</rosparam>
        </node>
    </group>

    <node name="control" pkg="drone_optimal_control" type="drone_control" output="screen"
          launch-prefix="taskset -c 3">
        <param name="use_fixed_guidance" value="$(arg use_fixed_guidance)"/>
        <param name="use_tracking_controller" value="$(eval arg('use_tracking_controller')!='NONE')"/>
        <rosparam param="target_apogee">[0,0,1]</rosparam>
        <rosparam param="track_guidance" subst_value="true">$(eval arg('use_guidance') or arg('use_fixed_guidance'))</rosparam>
        <param name="use_ground_truth_state" value="$(arg use_ground_truth_state)"/>
    </node>
</launch>
